tests/unit/service/test_base_document.py

```diff
@@ -551,3 +574,49 @@ async def test_non_auditable_document_service(non_auditable_document_service):
 
     audits = await service.list_document_audits(document_id=document.id)
     assert audits["total"] == 0
+
+
+@pytest.mark.asyncio
+async def test_app_container__disable_block_modification_check(app_container, entity):
+    """
+    Test app_container's disable_block_modification_check
+    """
+    # add block modification by to entity object
+    await app_container.entity_service.add_block_modification_by(
+        query_filter={"_id": entity.id},
+        reference_info=ReferenceInfo(asset_name="Asset", document_id=ObjectId()),
+    )
+
+    # check that entity is blocked from modification
+    with pytest.raises(DocumentModificationBlockedError):
+        await app_container.entity_service.update_document(
+            document_id=entity.id, data=EntityServiceUpdate(name="new_name")
+        )
+
+    # case 1: use context manager without using the yielded app_container
+    with app_container.disable_block_modification_check():
+        service = app_container.entity_service
+        updated_entity = await app_container.entity_service.update_document(
+            document_id=entity.id, data=EntityServiceUpdate(name="new_name")
+        )
+        assert updated_entity.name == "new_name"
+
+        # inside the context manager, check that block modification check is disabled
+        assert service._check_block_modification_func() is False
+
+    # outside the context manager, check that block modification check is enabled
+    assert service._check_block_modification_func() is True
+
+    # case 2: use context manager with using the yielded app_container
+    with app_container.disable_block_modification_check() as updated_app_container:
```

Comment:
Is this equivalent to case 1 above? Maybe can consider only supporting the `with app_container.disable_block_modification_check()` for a simpler interface, i.e. not yielding anything in the context manager since it actually modifies the original instance in place.

Reply:
Both cases are supported here. The test here just to check both cases.